### DISCUSSION TITLE
Can't run QT app.  (spyder 5.1.5 requires pyqtwebengine<5.13)

### DISCUSSION THREAD

**Discussion by mccoyn** [mjhoptics/ray-optics#81] [Q&A] [answered]
When I try to run the QT app, it automatically closes without showing anything.  It also generates crash dump files for pythonw.exe.  Any ideas on how to fix this?

When I installed rayoptics I got this error, which may be related.

> ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
> spyder 5.1.5 requires pyqtwebengine<5.13, which is not installed.

I tried installing PyQtWebEngine and got similar errors.

>D:\Temp>C:\ProgramData\Anaconda3\python -m pip install PyQtWebEngine
>
>Collecting PyQtWebEngine
>  Using cached PyQtWebEngine-5.15.5-cp36-abi3-win_amd64.whl (181 kB)
>Requirement already satisfied: PyQt5>=5.15.4 in c:\programdata\anaconda3\lib\site-packages (from PyQtWebEngine) (5.15.6)
>Requirement already satisfied: PyQt5-sip<13,>=12.8 in c:\programdata\anaconda3\lib\site-packages (from PyQtWebEngine) (12.9.1)
>Requirement already satisfied: PyQtWebEngine-Qt5>=5.15.2 in c:\programdata\anaconda3\lib\site-packages (from PyQtWebEngine) (5.15.2)
>Requirement already satisfied: PyQt5-Qt5>=5.15.2 in c:\programdata\anaconda3\lib\site-packages (from PyQt5>=5.15.4->PyQtWebEngine) (5.15.2)
>Installing collected packages: PyQtWebEngine
>ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source >of the following dependency conflicts.
>spyder 5.1.5 requires pyqt5<5.13, but you have pyqt5 5.15.6 which is incompatible.
>spyder 5.1.5 requires pyqtwebengine<5.13, but you have pyqtwebengine 5.15.5 which is incompatible.
>Successfully installed PyQtWebEngine-5.15.5

**Comment by mjhoptics** [comment]
Hello Nick,
I use spyder also and find that it's very picky with it's dependencies. IIRC it requires pyqt <5.13. What I usually do is install spyder first, which pulls in a ton of dependencies, and then I install ray-optics. This then requires far fewer dependencies to be loaded and everything seems to be happy.
I also use conda, which claims to have better dependency management than pip, but this is still a problem.
Hope this helps.
Mike

**Comment by mccoyn** [comment, marked as answer]
I had to activate the environment first.  I just had to run it from the command window that was created by the "Anaconda Prompt" link in the start menu.

I also found that I could get more information from the error by running "python -m rayoptics.exe" from a command window.

I did also run the following two lines, but I'm not sure if they were necessary.  Its been a while since I've done anything with Python so I'm warming up to this a bit.
> python -m pip install pyqtwebengine==5.12.1
> python -m pip install pyqt5==5.12.3
